kube-router --cleanup fails

I0727 11:34:09.082403   80883 network_policy_controller.go:696] Cleaning up iptables configuration permanently done by kube-router
I0727 11:34:09.090881   80883 network_policy_controller.go:774] Successfully cleaned the iptables configuration done by kube-router
I0727 11:34:09.091020   80883 network_services_controller.go:769] Cleaning up IPVS configuration permanently
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x194601a]

goroutine 1 [running]:
github.com/cloudnativelabs/kube-router/app/controllers.(*NetworkServicesController).Cleanup(0xc4206bddd0)
	/home/kube/go/src/github.com/cloudnativelabs/kube-router/app/controllers/network_services_controller.go:770 +0x6a
github.com/cloudnativelabs/kube-router/app.CleanupConfigAndExit()
	/home/kube/go/src/github.com/cloudnativelabs/kube-router/app/server.go:50 +0x91
main.main()
	/home/kube/go/src/github.com/cloudnativelabs/kube-router/kube-router.go:37 +0x288

--- a/app/controllers/network_services_controller.go
+++ b/app/controllers/network_services_controller.go
@@ -767,7 +767,14 @@ func getKubeDummyInterface() (netlink.Link, error) {
 func (nsc *NetworkServicesController) Cleanup() {
 	// cleanup ipvs rules by flush
 	glog.Infof("Cleaning up IPVS configuration permanently")
-	err := h.Flush()
+
+	handle, err := libipvs.New()
+	if err != nil {
+		glog.Errorf("Failed to cleanup ipvs rules: ", err.Error())
+		return
+	}
+
+	err = handle.Flush()
 	if err != nil {
 		glog.Errorf("Failed to cleanup ipvs rules: ", err.Error())
 		return

